### PR TITLE
PV Encryption via User Interface

### PR DESCRIPTION
**Pull request by am-agrawa**
This PR creates an encrypted storage class via UI, creates multiple PVCs using that encrypted storage class, run IOs, validates and deletes the resources created during the test.

### CODE REVIEW

File: tests/ui/test_pv_encryption_ui.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+                    key=vol_handle, path=self.vault.vault_backend_path
+                ):
+                    logger.info(f"Vault: Key deleted for {vol_handle}")
+                else:
+                    raise KMSResourceCleaneupError(
+                        f"Vault: Key deletion failed for {vol_handle}"
+                    )
+        # Deleting Storage Class via UI
+        logger.info("Deleting Storage Class via UI")
+        sc_deletion_check = sc_obj.delete_rbd_storage_class(sc_name=sc_name)
```
Comment (shyRozen) [inline tests/ui/test_pv_encryption_ui.py:295]: If test fails before that storageclass won't be deleted. For this case I've wrote the storageclass ui factory.
You are moving a way from a best practice implementation. 
All you got to do is copy the option of the function `create_encrypted_storage_class_ui` in the class to the factory, check in the factory is kms_address exist and send from there the option to build the storageclass with your new method `create_encrypted_storage_class_ui`.
This way you don't have to handle the teardown and it open to more options to work with a new block pool. 
We can do it together if you want.

Comment (am-agrawa) [reply tests/ui/test_pv_encryption_ui.py]: Done

File: ocs_ci/ocs/ui/storageclass.py [hunk 2]
```diff
@@ -91,3 +95,144 @@ def delete_rbd_storage_class(self, sc_name):
...
+        provisioner="rbd",
+        vol_binding_mode="Immediate",
+        service_name=None,
+        kms_address=None,
+        tls_server_name=None,
+    ):
+        """
+        Test for creation of storage class with encryption via UI
+        Args:
+                backend_path (str): name of the vault backend path
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/ui/storageclass.py]: Done

**Comment by am-agrawa** [comment]
Test passed on Jenkins run for OCS 4.8- 
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/17371/testReport/

**Review by ebenahar — DISMISSED** [review]

File: tests/ui/test_pv_encryption_ui.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    argnames=["kv_version"],
+    argvalues=[
+        pytest.param("v1", marks=pytest.mark.polarion_id("OCS-2585")),
+        pytest.param("v2", marks=pytest.mark.polarion_id("OCS-2592")),
+    ],
```
Comment (ebenahar) [inline tests/ui/test_pv_encryption_ui.py:41]: it seems like these are already exsisting polarion IDs - TestRbdPvEncryption

Comment (am-agrawa) [reply tests/ui/test_pv_encryption_ui.py:41]: @ebenahar That's true, so shouldn't we use them?

Comment (ebenahar) [reply tests/ui/test_pv_encryption_ui.py:41]: Since these are new tests we should have new Polarion IDs

Comment (am-agrawa) [reply tests/ui/test_pv_encryption_ui.py]: @ebenahar Done, pls approve once more.

**Comment by am-agrawa** [comment]
@shyRozen @rachael-george @OdedViner @shylesh Can I get review/approval pls?

**Comment by rachael-george** [comment]
Was this PR tested with ODF 4.12 and/or ODF 4.9 and above?

**Comment by am-agrawa** [comment]
> Was this PR tested with ODF 4.12 and/or ODF 4.9 and above?

No, that's why skip is added for these versions. Will take them in separate PRs.

**Review by rachael-george — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**